pkg/typeutil/string_slice.go

```diff
@@ -34,6 +34,10 @@ func (s *StringSlice) UnmarshalJSON(text []byte) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if len(data) == 0 {
+		*s = nil
```

Comment:
so if a slice has data before but be unmarshaled with an empty data, the slice may become nil, is it safe?